server.go

```diff
@@ -2715,6 +2715,48 @@ func (s *server) genNodeAnnouncement(refresh bool,
 	return *s.currentNodeAnn, nil
 }
 
+// updateAndBrodcastSelfNode generates a new node announcement
+// applying the giving modifiers and updating the time stamp
+// to ensure it propagates through the network. Then it brodcasts
+// it to the network.
+func (s *server) updateAndBrodcastSelfNode(
+	modifiers ...netann.NodeAnnModifier) error {
+
+	newNodeAnn, err := s.genNodeAnnouncement(true, modifiers...)
+	if err != nil {
+		return fmt.Errorf("unable to generate new node "+
+			"announcement: %v", err)
+	}
+
+	// Update the on-disk version of our announcement.
+	selfNode := &channeldb.LightningNode{
```

Comment:
I'm not sure if we should instead load the self node here first (`s.graphDB.SourceNode()`) and then overwrite the new fields? Otherwise we risk overwriting existing values (such as the `ExtraOpaqueData` field).

Reply:
Good point